### PR TITLE
Allow to show track details

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we introduce a new `info` subcommand to Rally that
allows users to print details about a track on the command line. This
can be useful to see which challenges exist and which tasks are executed
in each of them.

### CODE REVIEW

File: docs/adding_tracks.rst [hunk 1]
```diff
@@ -202,6 +202,35 @@ The new track appears when you run ``esrally list tracks --track-path=~/rally-tr
     ----------  ----------------------------- -----------  ---------------  -----------------
     tutorial    Tutorial benchmark for Rally      11658903  N/A              1.4 GB
 
+You can also show details about your track with ``esrally info --track-path=~/rally-tracks/tutorial``::
+
+    dm@io:~ $ esrally info --track-path=~/rally-tracks/tutorial
```
Comment (ebadyano) [inline docs/adding_tracks.rst]: should we remove `dm@io $` portion?

Comment (danielmitterdorfer) [reply docs/adding_tracks.rst]: This was kind of intentional to make it clear that this is a prompt.

**Comment by ebadyano** [comment]
Very cool feature, thank you! 

Do you think we want to add any tests? The feature itself is kind of like for testing, but maybe we want to add some to make sure some changes in the future don't break it?

**Comment by danielmitterdorfer** [comment]
> Do you think we want to add any tests?

Thanks for the review. As discussed on another channel, let's add some integration tests for starters.

**Comment by danielmitterdorfer** [comment]
@ebadyano I've added an integration test now in https://github.com/elastic/rally/pull/850/commits/2b0cb8e74e574bd31c009e10d500ad44005c168f. Can you please have another look?

**Comment by danielmitterdorfer** [comment]
Thanks for the review! :)

**Pull request merged**